Subject: [PATCH] Black screen for magnification gesture when play video with HWC

the root cause is computing source crop reduce transparent region
but not for visible region screen when using HWComposer to do composite.
use visibleNonTransparentRegion instead of visibleRegion to set
visible region screen to HWComposer.

diff --git a/services/surfaceflinger/Layer.cpp b/services/surfaceflinger/Layer.cpp
index 61af51f..30cf00c 100644
--- a/services/surfaceflinger/Layer.cpp
+++ b/services/surfaceflinger/Layer.cpp
@@ -806,7 +806,8 @@
     // Apply this display's projection's viewport to the visible region
     // before giving it to the HWC HAL.
     const Transform& tr = hw->getTransform();
-    Region visible = tr.transform(visibleRegion.intersect(hw->getViewport()));
+    Region visible = tr.transform(visibleNonTransparentRegion.intersect(
+                                   hw->getViewport()));
     layer.setVisibleRegionScreen(visible);
     layer.setSurfaceDamage(surfaceDamageRegion);
     mIsGlesComposition = (layer.getCompositionType() == HWC_FRAMEBUFFER);
